CHANGELOG.md

```diff
@@ -1,5 +1,113 @@
 # TigerBeetle Changelog
 
+## 2024-07-26
+
+### Safety And Performance
+
+- [#2140](https://github.com/tigerbeetle/tigerbeetle/pull/2140)
+
+  Fix a bug where MessageBus sees block messages (due to state sync or repair) and peer_type says
+  they are always from replica 0 (since Header.Block.replica == 0 always). So, if they are being
+  sent by a non-R0 replica, it drops the messages with message from unexpected peer.
```

Comment:
```suggestion
  sent by a non-R0 replica, it drops the messages with "message from unexpected peer".
```

Also motivate the ultimate impact, not only that messages are dropped, but that this results in a replica stuck in state sync.